client/allocrunner/taskrunner/template_hook.go

```diff
@@ -113,6 +121,20 @@ func (h *templateHook) Prestart(ctx context.Context, req *interfaces.TaskPrestar
 	h.vaultToken = req.VaultToken
 	h.nomadToken = req.NomadToken
 
+	// Set the consul token if the task uses WI
+	if req.Task.Consul != nil {
+		consulTokens := h.config.hookResources.GetConsulTokens()
+
+		var found bool
+		h.consulToken, found = consulTokens[req.Task.Consul.Cluster][req.Task.Consul.IdentityName()]
+		if !found {
+			return fmt.Errorf(
+				"consul task %s uses workload identity, but unable to find a consul token for that task",
+				req.Task.Name,
+			)
+		}
```

Comment:
We're returning an error here because we (rightly, I think) want to protect ourselves if something isn't quite right here. But if that's the case, maybe we should ok-check/nil-check after `consulTokens[req.Task.Consul.Cluster]` too? Just in case that's where the problem is?